Production/govuk_ios/Helpers/StringValidator/ValidatorProvider.swift

```diff
@@ -0,0 +1,40 @@
+import Foundation
+
+protocol ValidatorProvider {
+    func validate(input: String?) -> Bool
+}
+
+struct MaximumLengthValidator: ValidatorProvider {
+    private let requiredStringLength: Int
+
+    init(requiredStringLength: Int) {
+        self.requiredStringLength = requiredStringLength
+    }
+
+    func validate(input: String?) -> Bool {
+        guard let input = input else { return false }
+        return input.count <= requiredStringLength
+    }
+}
+
+struct MinimumLengthValidator: ValidatorProvider {
+    private let requiredStringLength: Int
+
+    init(requiredStringLength: Int) {
+        self.requiredStringLength = requiredStringLength
+    }
+
+    func validate(input: String?) -> Bool {
+        guard let input = input else { return false }
+        return input.count >= requiredStringLength
+    }
+}
+
+struct CharValidator: ValidatorProvider {
+    func validate(input: String?) -> Bool {
+        guard let input = input else { return false }
+        return input.range(
+            of: ".*[^A-Za-z0-9].*",
```

Comment:
This is quite aggressive validation. Also, doesn't quite meet the requirements.
"Validates there are no invalid chars in the input" specifically doesn't state what an invalid char is.
If it helps for the purposes of the PR, lets say invalid chars are `*`, `-` and `{`